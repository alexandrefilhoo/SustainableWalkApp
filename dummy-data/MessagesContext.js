// Random contact messages
const dummyMessages = [

  {
    id: "1",
    name: "Aline Thomas",
    degree: "MSc Medicine",
    university: "King's College London",
    image: require("../images/person-14.webp"),
    time: "",
    message: "",
  },
  {
    id: "2",
    name: "Clarisse Alves",
    degree: "MSc Biomedicine",
    university: "Imperial College London",
    image: require("../images/person-2.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "3",
    name: "Barbara Johnson",
    degree: "BSc Entrepreneurship",
    university: "University College London",
    image: require("../images/person-4.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "4",
    name: "Paul Lewis",
    degree: "MSc Medicine",
    university: "University of Birmingham",
    image: require("../images/person-3.jpeg"),
    time: "",
    message: "",
      
  },
  {
    id: "5",
    name: "Elton Thomas",
    degree: "BSc Graphic Design",
    university: "University of Westminster",
    image: require("../images/person-18.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "6",
    name: "Amanda Jones",
    degree: "BSc Computer Science",
    university: "University of Glasgow",
    image: require("../images/person-9.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "7",
    name: "Wei Li ",
    degree: "BSc Archaeology",
    university: "University College London",
    image: require("../images/person-20.jpeg"),
    time: "",
    message: "",
      
  },
  {
    id: "8",
    name: "Alexandra Moore",
    degree: "MSc Data Science",
    university: "King’s  College London",
    image: require("../images/person-21.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "9",
    name: "Stefano Cipriani ",
    degree: "BSc Entrepreneurship",
    university: "Imperial College London",
    image: require("../images/person-23.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "10",
    name: "Min Zhou",
    degree: "BSc Business Administration",
    university: "King’s College London",
    image: require("../images/person-22.jpeg"),
    time: "",
    message: "",
      
  },
  {
    id: "11",
    name: "Raj Patel",
    degree: "BSc Computer Science",
    university: "University of Oxford",
    image: require("../images/person-26.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "12",
    name: "Omar Rohan",
    degree: "MSc Artificial Intelligence",
    university: "Imperial College London",
    image: require("../images/person-25.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "13",
    name: "Sara Willians",
    degree: "MSc Environmetal Engineering",
    university: "Imperial College London",
    image: require("../images/person-28.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "14",
    name: "Clare Giordano",
    degree: "BSc Psychology ",
    university: "University of Greenwich",
    image: require("../images/person-29.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "15",
    name: "Samantha Ferrari",
    degree: "BSc  Geology",
    university: "University of Birmingham",
    image: require("../images/person-30.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "16",
    name: "Sofia Turner",
    degree: "BSc Law",
    university: "Queen Mary, University of London",
    image: require("../images/person-31.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "17",
    name: "Steven Harris",
    degree: "MSc Biomedicine",
    university: "Imperial  College London",
    image: require("../images/person-36.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "18",
    name: "Sasha Bianchi",
    degree: "BSc Entrepreneurship",
    university: "King’s College London",
    image: require("../images/person-32.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "19",
    name: "Pedro Sanchez",
    degree: "BSc Computer Science",
    university: "City, University of London",
    image: require("../images/person-13.png"),
    time: "",
    message: "",
  },
  {
    id: "20",
    name: "Lucia Menotti",
    degree: "BSc Chemical Engineering",
    university: "University of Bath",
    image: require("../images/person-11.jpg"),
    time: "",
    message: "",
  },
  {
    id: "21",
    name: "Barbara Robbins",
    degree: "BSc Architecture",
    university: "University of Cambridge",
    image: require("../images/person-27.jpg"),
    time: "",
    message: "",
  },
  {
    id: "22",
    name: "Min Zhang",
    degree: "BSc Geology",
    university: "University of Bath",
    image: require("../images/person-50.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "23",
    name: "Alicia Schmidt ",
    degree: "BSc Physics",
    university: "University of  Sunderland",
    image: require("../images/person-33.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "24",
    name: "Steph  Robbins",
    degree: "BSc Politics",
    university: "Queen Mary, University of London",
    image: require("../images/person-52.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "25",
    name: "Chris  Heritage",
    degree: "MSc Architecture ",
    university: "University of East London",
    image: require("../images/person-55.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "26",
    name: "Iago Bocelli",
    degree: "BSc Photography",
    university: "University of Arts, London",
    image: require("../images/person-39.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "27",
    name: "Amanda Hoffmann",
    degree: "BSc Fashion",
    university: "University of Arts, London",
    image: require("../images/person-47.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "28",
    name: "Steven Isaac ",
    degree: "BSc Cyber Security ",
    university: "City, University of London",
    image: require("../images/person-56.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "29",
    name: "Fang Yang ",
    degree: "BSc Petroleum Engineering",
    university: "Imperial College London",
    image: require("../images/person-57.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "30",
    name: "Caroline  Queiroz",
    degree: "BSc Public Relations",
    university: "University of Birgminham",
    image: require("../images/person-58.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "31",
    name: "Stella Artois",
    degree: "BSc Business Management ",
    university: "London School of Economics",
    image: require("../images/person-59.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "32",
    name: "Ji-Hoon Jang",
    degree: "BSc Law",
    university: "King’s College London",
    image: require("../images/person-60.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "33",
    name: "Vicky Simon",
    degree: "BSc Chemistry",
    university: "Imperial College London",
    image: require("../images/person-61.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "34",
    name: "Claudia Benitez",
    degree: "BSc Computer Science",
    university: "City, University of London",
    image: require("../images/person-62.jpg"),
    time: "",
    message: "",
    
  },
  {
    id: "35",
    name: "Sandro Andre",
    degree: "BSc Chemical Engineering",
    university: "University of Bath",
    image: require("../images/person-63.jpg"),
    time: "",
    message: "",
      
  },
  {
    id: "36",
    name: "Alexander Miller",
    degree: "BSc Computer Science",
    university: "University of Cambridge",
    image: require("../images/person-64.jpg"),
    time: "",
    message: "",
      
  },
];

export default dummyMessages;
