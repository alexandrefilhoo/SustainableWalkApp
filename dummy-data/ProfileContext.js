/**
 * Code to create a contex for managingg user profile. Initilizes the state 
 * for the profile, and provide functions to updte the profile.
 */
import React, { createContext, useState, useContext } from "react";

const ProfileContext = createContext();

export const ProfileProvider = ({ children }) => {

  // Using useState hook initilize the state for the profile
  const [profile, setProfile] = useState({
    name: "",
    email: "",
    university: "",
    degreeTitle: "",
    dob: "",
    phone: "",
    profileImage: null,
  });

  const [theme, setTheme] = useState("light");

  // Updates profile
  const updateProfile = (newProfile) => {
    setProfile((prevProfile) => ({ ...prevProfile, ...newProfile }));
  };

  return (
    <ProfileContext.Provider
      value={{ profile,setProfile, updateProfile, theme, setTheme }}
    >
      {children}
    </ProfileContext.Provider>
  );
};

export const useProfile = () => useContext(ProfileContext);
