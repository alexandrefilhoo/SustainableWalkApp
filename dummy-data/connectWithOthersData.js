// This code is not being used for the applicatinon yet. Code for future app expansion.
export const peopleYouMayKnow = [
  {
    id: "1",
    name: "Aline Thomas",
    degree: "MSc Medicine",
    university: "King's College London",
    image: require("../images/person-14.webp"),
    bio: "I'm pursuing a Master's degree in Medicine at King's College London, focusing on clinical practice and medical research. With a passion for improving patient care and a strong foundation in biomedical sciences, I have developed a keen interest in cardiology. My goal is to contribute meaningfully to the medical community and enhance healthcare quality. Outside academics, I enjoy jogging to maintain a balanced lifestyle.",
  },
  {
    id: "2",
    name: "Clarisse Alves",
    degree: "MSc Biomedicine",
    university: "Imperial College London",
    image: require("../images/person-2.jpg"),
    bio: "",
  },
  {
    id: "3",
    name: "Barbara Johnson",
    degree: "BSc Entrepreneurship",
    university: "University College London",
    image: require("../images/person-4.jpg"),
    bio: "",
  },
  {
    id: "4",
    name: "Paul Lewis",
    degree: "MSc Medicine",
    university: "University of Birmingham",
    image: require("../images/person-3.jpeg"),
    bio: "",
  },
  {
    id: "5",
    name: "Elton Thomas",
    degree: "BSc Graphic Design",
    university: "University of Westminster",
    image: require("../images/person-18.jpg"),
    bio: "",
  },
  {
    id: "6",
    name: "Amanda Jones",
    degree: "BSc Computer Science",
    university: "University of Glasgow",
    image: require("../images/person-9.jpg"),
    bio: "",
  },
  {
    id: "7",
    name: "Wei Li ",
    degree: "BSc Archaeology",
    university: "University College London",
    image: require("../images/person-20.jpeg"),
    bio: "",
  },
  {
    id: "8",
    name: "Alexandra Moore",
    degree: "MSc Data Science",
    university: "King’s  College London",
    image: require("../images/person-21.jpg"),
    bio: "",
  },
  {
    id: "9",
    name: "Stefano Cipriani ",
    degree: "BSc Entrepreneurship",
    university: "Imperial College London",
    image: require("../images/person-23.jpg"),
    bio: "",
  },
  {
    id: "10",
    name: "Min Zhou",
    degree: "BSc Business Administration",
    university: "King’s College London",
    image: require("../images/person-22.jpeg"),
    bio: "",
  },
  {
    id: "11",
    name: "Raj Patel",
    degree: "BSc Computer Science",
    university: "University of Oxford",
    image: require("../images/person-26.jpg"),
    bio: "",
  },
  {
    id: "12",
    name: "Omar Rohan",
    degree: "MSc Artificial Intelligence",
    university: "Imperial College London",
    image: require("../images/person-25.jpg"),
    bio: "",
  },
  {
    id: "13",
    name: "Sara Willians",
    degree: "MSc Environmetal Engineering",
    university: "Imperial College London",
    image: require("../images/person-28.jpg"),
    bio: "",
  },
  {
    id: "14",
    name: "Clare Giordano",
    degree: "BSc Psychology ",
    university: "University of Greenwich",
    image: require("../images/person-29.jpg"),
    bio: "",
  },
  {
    id: "15",
    name: "Samantha Ferrari",
    degree: "BSc  Geology",
    university: "University of Birmingham",
    image: require("../images/person-30.jpg"),
    bio: "",
  },
  {
    id: "16",
    name: "Sofia Turner",
    degree: "BSc Law",
    university: "Queen Mary, University of London",
    image: require("../images/person-31.jpg"),
    bio: "",
  },
  {
    id: "17",
    name: "Steven Harris",
    degree: "MSc Biomedicine",
    university: "Imperial  College London",
    image: require("../images/person-36.jpg"),
    bio: "",
  },
  {
    id: "18",
    name: "Sasha Bianchi",
    degree: "BSc Entrepreneurship",
    university: "King’s College London",
    image: require("../images/person-32.jpg"),
    bio: "",
  },
  {
    id: "19",
    name: "Pedro Sanchez",
    degree: "BSc Computer Science",
    university: "City, University of London",
    image: require("../images/person-13.png"),
  },
  {
    id: "20",
    name: "Lucia Menotti",
    degree: "BSc Chemical Engineering",
    university: "University of Bath",
    image: require("../images/person-11.jpg"),
  },
  {
    id: "21",
    name: "Barbara Robbins",
    degree: "BSc Architecture",
    university: "University of Cambridge",
    image: require("../images/person-27.jpg"),
  },
  {
    id: "22",
    name: "Min Zhang",
    degree: "BSc Geology",
    university: "University of Bath",
    image: require("../images/person-50.jpg"),
    bio: "",
  },
  {
    id: "23",
    name: "Alicia Schmidt ",
    degree: "BSc Physics",
    university: "University of  Sunderland",
    image: require("../images/person-33.jpg"),
    bio: "",
  },
  {
    id: "24",
    name: "Steph  Robbins",
    degree: "BSc Politics",
    university: "Queen Mary, University of London",
    image: require("../images/person-52.jpg"),
    bio: "",
  },
  {
    id: "25",
    name: "Chris  Heritage",
    degree: "MSc Architecture ",
    university: "University of East London",
    image: require("../images/person-55.jpg"),
    bio: "",
  },
  {
    id: "26",
    name: "Iago Bocelli",
    degree: "BSc Photography",
    university: "University of Arts, London",
    image: require("../images/person-39.jpg"),
    bio: "",
  },
  {
    id: "27",
    name: "Amanda Hoffmann",
    degree: "BSc Fashion",
    university: "University of Arts, London",
    image: require("../images/person-47.jpg"),
    bio: "",
  },
  {
    id: "28",
    name: "Steven Isaac ",
    degree: "BSc Cyber Security ",
    university: "City, University of London",
    image: require("../images/person-56.jpg"),
    bio: "",
  },
  {
    id: "29",
    name: "Fang Yang ",
    degree: "BSc Petroleum Engineering",
    university: "Imperial College London",
    image: require("../images/person-57.jpg"),
    bio: "",
  },
  {
    id: "30",
    name: "Caroline  Queiroz",
    degree: "BSc Public Relations",
    university: "University of Birgminham",
    image: require("../images/person-58.jpg"),
    bio: "",
  },
  {
    id: "31",
    name: "Stella Artois",
    degree: "BSc Business Management ",
    university: "London School of Economics",
    image: require("../images/person-59.jpg"),
    bio: "",
  },
  {
    id: "32",
    name: "Ji-Hoon Jang",
    degree: "BSc Law",
    university: "King’s College London",
    image: require("../images/person-60.jpg"),
    bio: "",
  },
  {
    id: "33",
    name: "Vicky Simon",
    degree: "BSc Chemistry",
    university: "Imperial College London",
    image: require("../images/person-61.jpg"),
    bio: "",
  },
  {
    id: "34",
    name: "Claudia Benitez",
    degree: "BSc Computer Science",
    university: "City, University of London",
    image: require("../images/person-62.jpg"),
    bio: "",
  },
  {
    id: "35",
    name: "Sandro Andre",
    degree: "BSc Chemical Engineering",
    university: "University of Bath",
    image: require("../images/person-63.jpg"),
    bio: "",
  },
  {
    id: "36",
    name: "Alexander Miller",
    degree: "BSc Computer Science",
    university: "University of Cambridge",
    image: require("../images/person-64.jpg"),
    bio: "",
  },
];
