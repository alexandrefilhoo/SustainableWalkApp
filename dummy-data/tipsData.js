/**
 * Tips and suggestions data. Each tips contains a title, image, category, details, and tags
 */
export const tipsData = [
  {
    title: "Use Public Transport",
    image: require("../images/use-public-transport.jpg"),
    category: "Tips for Sustainable Transportation",
    details:
      "Opt for buses, trains, trams, or subways instead of personal vehicles. Public transport reduces the number of vehicles on the road, decreasing emissions and congestion.",
    tags: ["Public Transport", "Reduce Emissions", "Reduce Congestion"],
  },
  {
    title: "Carpooling",
    image: require("../images/carpooling.jpg"),
    category: "Tips for Sustainable Transportation",
    details:
      "Share rides with friends, family, or colleagues. Carpooling reduces the number of cars on the road, leading to lower greenhouse gas emissions.",
    tags: ["Share Rides", "Reduce Cars", "Lower Emissions"],
  },
  {
    title: "Biking",
    image: require("../images/biking.jpg"),
    category: "Tips for Sustainable Transportation",
    details:
      "Use bicycles for short-distance travel. Biking is a zero-emission mode of transport and offers health benefits.",
    tags: ["Health benefits", "Short-distance travel", "Zero-Emission"],
  },
  {
    title: "Telecommuting",
    image: require("../images/telecommuting.jpg"),
    category: "Tips for Sustainable Transportation",
    details:
      "Work from home when possible to reduce commuting frequency. Telecommuting cuts down on transportation emissions.",
    tags: ["Remote Work", "Reduce Commuting", "Telecommute"],
  },
  {
    title: "Eco-Driving Techniques",
    image: require("../images/eco-friendly-driving.jpg"),
    category: "Tips for Sustainable Transportation",
    details:
      "Drive smoothly, avoid sudden accelerations and decelerations, and maintain steady speeds. Properly maintain your vehicle to ensure it runs efficiently.",
    tags: ["Smooth Driving", "Steady speeds", "Efficient Driving"],
  },
  {
    title: "Electric Vehicles",
    image: require("../images/eletric-vehicle.jpg"),
    category: "Tips for Sustainable Transportation",
    details:
      "Share rides with friends, family, or colleagues. Carpooling reduces the number of cars on the road, leading to lower greenhouse gas emissions.",
    tags: ["Green Cars", "Fewer Emissions", "Hybrid/Electric vehicles"],
  },
  {
    title: "Short Distance Walking",
    image: require("../images/walking-for-short-distance.jpeg"),
    category: "Tips for Encouraging Walking",
    details:
      "For distances under 3 kilometres, opt to walk instead of driving. Walking reduces emissions and contributes to personal fitness.",
    tags: ["Reduce Emissions", "Zero Emission", "Personal Fitness"],
  },
  {
    title: "Walk to Public Transport",
    image: require("../images/walk-to-public-transport.jpg"),
    category: "Tips for Encouraging Walking",
    details:
      "Combine walking with public transportation. Walking to and from transit stops can be an easy way to incorporate more physical activity into your day.",
    tags: ["Combine Walking", "Physical Activity", "Sustainable Commute"],
  },
  {
    title: "Walking Programs",
    image: require("../images/walk-campaign.png"),
    category: "Tips for Encouraging Walking",
    details:
      "Participate in or organize community walking groups or events to promote walking as a fun and social activity.",
    tags: ["Community Walking", "Walking Groups", "Physical Activity"],
  },
  {
    title: "Awareness Campaigns",
    image: require("../images/community-walk-program.jpg"),
    category: "Tips for Encouraging Walking",
    details:
      "Support awareness campaigns that highlight the benefits of walking and the impact of reduced car use on the environment.",
    tags: ["Reduce car use", "Environmental Impact", "Promote Walking"],
  },
  {
    title: "Raise Awareness",
    image: require("../images/educate-and-raise-awareness.png"),
    category: "General Tips",
    details:
      "Inform others about the environmental and health benefits of sustainable transportation and walking. ",
    tags: ["Educate Others", "Community Action", "Awareness Campaigns"],
  },
  {
    title: "Use Technology",
    image: require("../images/use-technology.jpg"),
    category: "General Tips",
    details:
      "Use apps and online tools to find the best transport routes, carpooling opportunities, and biking paths. Track your walking and biking distances to stay motivated.",
    tags: ["Online Tools", "Track Distance", "Public Transport Routes"],
  },
  {
    title: "Reduce Travel Needs",
    image: require("../images/reduce-travel-needs.jpg"),
    category: "General Tips",
    details:
      "Plan and combine errands to minimize trips. Online shopping and delivery services can also reduce the need for multiple car trips.",
    tags: ["Minimize Trips", "Online Shopping", "Delivery Services"],
  },
  {
    title: "Support Green Initiatives",
    image: require("../images/support-green-initiatives.jpg"),
    category: "General Tips",
    details:
      "Back initiatives that aim to reduce the carbon footprint of transportation, such as renewable energy projects and sustainable urban planning efforts.",
    tags: ["Green Initiaves", "Lower Emissions", "Ride-Sharing"],
  },
];
