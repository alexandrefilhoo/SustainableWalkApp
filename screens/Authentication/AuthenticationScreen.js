/**
 * Code for the user autentication, providing a user interface for logging in and registering.
 * Includes the input fields for email and password with validations, and buttons for login,
 * register and forgot password.
 */
import React, { useState } from "react";
import {
  View,
  Text,
  TextInput,
  StyleSheet,
  TouchableOpacity,
  ImageBackground,
  KeyboardAvoidingView,
  Platform,
  Alert,
} from "react-native";
import { useNavigation } from "@react-navigation/native";
import { useProfile } from "../../dummy-data/ProfileContext";


// Define the component using onAuthSuccess as the function prop.
const AuthenticationScreen = ({ onAuthSuccess }) => {
  const navigation = useNavigation();
  const { profile } = useProfile();
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [emailError, setEmailError] = useState("");
  const [passwordError, setPasswordError] = useState("");


  // Button press handler for login
  const handleLogin = () => {
    let valid = true;
    if (!email) {
      setEmailError("Enter email");
      valid = false;
    } else {
      setEmailError("");
    }
    if (!password) {
      setPasswordError("Enter password");
      valid = false;
    } else {
      setPasswordError("");
    }

    // Check if user has registered in the app
    if (valid) {
      if (profile.email === email) {
        onAuthSuccess();
      } else {
        Alert.alert("Error", "User not registered. Please register first.");
      }
    }
  };


  // Function to navigate to Registration Screen
  const handleRegister = () => {
    navigation.navigate("Registration");
  };

  // Function to navigate to Forgoto Password
  const handleForgotPassword = () => {
    navigation.navigate("ForgotPassword");
  };


  // Returns the Login and Register Inital Screen.
  return (
    <ImageBackground
      source={require("../../images/background-5.jpg")}
      style={styles.backgroundImage}
    >
      <KeyboardAvoidingView
        style={{ flex: 1 }}
        behavior={Platform.OS === "ios" ? "padding" : "height"}
        keyboardVerticalOffset={90}
      >
        <View style={styles.container}>
          <View style={styles.formContainer}>
            <TextInput
              style={styles.input}
              placeholder="Email"
              placeholderTextColor="#888"
              value={email}
              onChangeText={setEmail}
            />
            {emailError ? (
              <Text style={styles.errorText}>{emailError}</Text>
            ) : null}
            <TextInput
              style={styles.input}
              placeholder="Password"
              placeholderTextColor="#888"
              secureTextEntry
              value={password}
              onChangeText={setPassword}
            />
            {passwordError ? (
              <Text style={styles.errorText}>{passwordError}</Text>
            ) : null}
            <TouchableOpacity onPress={handleForgotPassword}>
              <Text style={styles.forgotPasswordText}>Forgot password ?</Text>
            </TouchableOpacity>
            <TouchableOpacity style={styles.loginButton} onPress={handleLogin}>
              <Text style={styles.loginButtonText}>Login</Text>
            </TouchableOpacity>
            <TouchableOpacity
              style={styles.registerButton}
              onPress={handleRegister}
            >
              <Text style={styles.registerButtonText}>Register</Text>
            </TouchableOpacity>
          </View>
        </View>
      </KeyboardAvoidingView>
    </ImageBackground>
  );
};

const styles = StyleSheet.create({
  backgroundImage: {
    flex: 1,
    resizeMode: "cover",
    justifyContent: "center",
  },
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  formContainer: {
    width: "80%",
    backgroundColor: "rgba(255, 255, 255, 0.9)",
    padding: 20,
    borderRadius: 10,
  },
  input: {
    height: 40,
    borderColor: "#ccc",
    borderBottomWidth: 1,
    marginBottom: 10,
    fontSize: 16,
    color: "#000",
  },
  errorText: {
    color: "red",
    fontSize: 14,
    marginBottom: 10,
  },
  forgotPasswordText: {
    color: "#2d572c",
    textAlign: "right",
    marginBottom: 20,
  },
  loginButton: {
    backgroundColor: "#2d572c",
    padding: 15,
    borderRadius: 10,
    alignItems: "center",
    marginBottom: 10,
  },
  loginButtonText: {
    color: "#fff",
    fontSize: 16,
    fontWeight: "bold",
  },
  registerButton: {
    backgroundColor: "#FDA701",
    padding: 15,
    borderRadius: 10,
    alignItems: "center",
  },
  registerButtonText: {
    color: "#fff",
    fontSize: 16,
    fontWeight: "bold",
  },
});

export default AuthenticationScreen;
