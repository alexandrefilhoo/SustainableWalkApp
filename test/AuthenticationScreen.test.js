import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import AuthenticationScreen from "../screens/Authentication/AuthenticationScreen";
import { useNavigation } from "@react-navigation/native";
import { useProfile } from "../dummy-data/ProfileContext";
import { Alert } from "react-native";

// Mock useNavigation and useProfile hooks
jest.mock("@react-navigation/native", () => ({
  useNavigation: jest.fn(),
}));

jest.mock("../dummy-data/ProfileContext", () => ({
  useProfile: jest.fn(),
}));

describe("AuthenticationScreen", () => {
  const mockNavigate = jest.fn();
  const mockOnAuthSuccess = jest.fn();

  beforeEach(() => {
    useNavigation.mockReturnValue({ navigate: mockNavigate });
    useProfile.mockReturnValue({ profile: { email: "test@example.com" } });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("shows error messages if email and password fields are empty", () => {
    const { getByText } = render(
      <AuthenticationScreen onAuthSuccess={mockOnAuthSuccess} />
    );

    fireEvent.press(getByText("Login"));

    expect(getByText("Enter email")).toBeTruthy();
    expect(getByText("Enter password")).toBeTruthy();
  });

  it("calls the method onAuthSuccess only if the email and password are precise", async () => {
    const { getByText, getByPlaceholderText } = render(
      <AuthenticationScreen onAuthSuccess={mockOnAuthSuccess} />
    );

    fireEvent.changeText(getByPlaceholderText("Email"), "test@example.com");
    fireEvent.changeText(getByPlaceholderText("Password"), "password");
    fireEvent.press(getByText("Login"));

    await waitFor(() => {
      expect(mockOnAuthSuccess).toHaveBeenCalled();
    });
  });


  it("navigates the user to the Registration screen by pressing the Register button", () => {
    const { getByText } = render(
      <AuthenticationScreen onAuthSuccess={mockOnAuthSuccess} />
    );

    fireEvent.press(getByText("Register"));

    expect(mockNavigate).toHaveBeenCalledWith("Registration");
  });

  it("navigates the user to the Forgot Password screen by pressing Forgot Password button", () => {
    const { getByText } = render(
      <AuthenticationScreen onAuthSuccess={mockOnAuthSuccess} />
    );

    fireEvent.press(getByText("Forgot password ?"));

    expect(mockNavigate).toHaveBeenCalledWith("ForgotPassword");
  });

  it("renders correctly", () => {
    const tree = renderer.create(<AuthenticationScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
