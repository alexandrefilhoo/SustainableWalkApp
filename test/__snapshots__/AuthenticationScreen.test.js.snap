// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticationScreen renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "flex": 1,
      "justifyContent": "center",
      "resizeMode": "cover",
    }
  }
>
  <Image
    source={1}
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    onLayout={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "rgba(255, 255, 255, 0.9)",
            "borderRadius": 10,
            "padding": 20,
            "width": "80%",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Email"
          placeholderTextColor="#888"
          style={
            {
              "borderBottomWidth": 1,
              "borderColor": "#ccc",
              "color": "#000",
              "fontSize": 16,
              "height": 40,
              "marginBottom": 10,
            }
          }
          value=""
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Password"
          placeholderTextColor="#888"
          secureTextEntry={true}
          style={
            {
              "borderBottomWidth": 1,
              "borderColor": "#ccc",
              "color": "#000",
              "fontSize": 16,
              "height": 40,
              "marginBottom": 10,
            }
          }
          value=""
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              {
                "color": "#2d572c",
                "marginBottom": 20,
                "textAlign": "right",
              }
            }
          >
            Forgot password ?
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#2d572c",
              "borderRadius": 10,
              "marginBottom": 10,
              "opacity": 1,
              "padding": 15,
            }
          }
        >
          <Text
            style={
              {
                "color": "#fff",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Login
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#FDA701",
              "borderRadius": 10,
              "opacity": 1,
              "padding": 15,
            }
          }
        >
          <Text
            style={
              {
                "color": "#fff",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Register
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
